core/src/main/java/bisq/core/api/CoreDisputeAgentService.java

```diff
@@ -0,0 +1,187 @@
+/*
+ * This file is part of Bisq.
+ *
+ * Bisq is free software: you can redistribute it and/or modify it
+ * under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or (at
+ * your option) any later version.
+ *
+ * Bisq is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE. See the GNU Affero General Public
+ * License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with Bisq. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package bisq.core.api;
+
+import bisq.core.support.dispute.mediation.mediator.Mediator;
+import bisq.core.support.dispute.mediation.mediator.MediatorManager;
+import bisq.core.support.dispute.refund.refundagent.RefundAgent;
+import bisq.core.support.dispute.refund.refundagent.RefundAgentManager;
+
+import bisq.network.p2p.NodeAddress;
+import bisq.network.p2p.P2PService;
+import bisq.network.p2p.P2PServiceListener;
+
+import bisq.common.config.Config;
+import bisq.common.crypto.KeyRing;
+
+import org.bitcoinj.core.ECKey;
+
+import javax.inject.Inject;
+
+import java.util.Arrays;
+import java.util.Date;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+
+import lombok.extern.slf4j.Slf4j;
+
+import static bisq.common.app.DevEnv.DEV_PRIVILEGE_PRIV_KEY;
+
+@Slf4j
+class CoreDisputeAgentService {
+
+    private final Config config;
+    private final KeyRing keyRing;
+    private final MediatorManager mediatorManager;
+    private final RefundAgentManager refundAgentManager;
+    private final P2PService p2PService;
+
+    @Inject
+    public CoreDisputeAgentService(Config config,
+                                   KeyRing keyRing,
+                                   MediatorManager mediatorManager,
+                                   RefundAgentManager refundAgentManager,
+                                   P2PService p2PService) {
+        this.config = config;
+        this.keyRing = keyRing;
+        this.mediatorManager = mediatorManager;
+        this.refundAgentManager = refundAgentManager;
+        this.p2PService = p2PService;
+        setupListeners();
+    }
+
+    public Optional<Mediator> getMediator(NodeAddress nodeAddress) {
+        return mediatorManager.getDisputeAgentByNodeAddress(nodeAddress);
+    }
+
+    public Optional<RefundAgent> getRefundAgent(NodeAddress nodeAddress) {
+        return refundAgentManager.getDisputeAgentByNodeAddress(nodeAddress);
+    }
+
+    private boolean shouldRegisterTestArbitrationDisputeAgents() {
```

Comment:
This seems odd, having all this test code in a non test class. Seems like part of this is useful for mainnet as well.

Reply:
Until I write the API methods for registering dispute agents, it can only be done in the UI.  The test harness starts up supporting apps (bitcoind,seednode,arbnode,etc) in separate linux processes, so test cases can only change state in those supporting Bisq apps via the gRPC API.  I'm doing a little triage here, and writing gRPC API support for registering dispute agents from a CLI was moved to the back of the line.  Maybe that's a mistake, but this is my reasoning...

This odd class was a short cut for automatically registering dispute agents for api test cases only.  I did not want to automatically register dispute agents in any daemon but a test arb node.  On mainnet, a user should use the CLI to do this; during testing, a dev (or Travis CI) should not have to run a UI to register dispute agents during the dev/test cycle.

This is a short cut to getting a "simplest trading script" ready for demonstration -- but well tested, and without having to do anything in a UI.  I did not think that an arbitrator CLI + daemon would be in the scope of the "simplest trading script" work (maybe that was a mistake), and I intended to remove this test related code when I do build out the API to support arbitrator CLI + daemons.

It took much more time figuring out when dispute agents can be registered than coding this class.  If I had known already, I could have thought this through better than I did, and built the gRPC API support for registering dispute agents from a CLI instead of this temporary short cut.  If it's not acceptable, I can build the API support next, and remove this test code.

Reply:
I don't think we should allow mainnet registration of dispute agent via cli for now. There are a lot of very specific dispute handling that I think is handled on the UI side. A UI of some sort would also be needed since it's a human facing role that's done manually.

For testing it's fine to have an automatically registered agent of course. Still would prefer if the test specifics were moved to the tests package.

Reply:
How about a new API method `registerdisputeagent [dispute-agent-type]` that does not work on mainnet?  It would throw an exception to a mainnet CLI, and the user would see an `Error: dispute agents must be registered in the Bisq UI` msg.

Headless regtest api test cases could use this method.

Sound OK?

Reply:
Yeah, sounds better. Let's go with that.

Reply:
The combined 9 commits below is a fix for the weird short cut I took for automatically registering dispute agents in api test cases.  Now it's done over gRPC with a `registerdisputeagent` method that will not work on mainnet.

It required some refactoring in the test harness so api calls could be made to an arbitration daemon, and there are a few other minor changes:  renaming a class, line formatting in the cli, and removing hard coded "localhost" strings.

Reply:
I have an unresolved file conflict, might need some help tomorrow.